src/middlewared/middlewared/plugins/iscsi_/targets.py

```diff
@@ -256,7 +257,9 @@ async def __validate(self, verrors, data, schema_name, old=None):
                             f'Authentication group {group["auth"]} does not support CHAP Mutual'
                         )
 
-    @accepts(Str('name'), Int('existing_id', null=True, default=None))
+    @accepts(Str('name'),
+             Int('existing_id', null=True, default=None),
+             roles=['SHARING_ISCSI_TARGET_READ'])
```

Comment:
Does `READ` role need this? Or only `WRITE` one?

Reply:
You are correct, even though this is a READ operation, it should only be needed by the `WRITE` role.  Thanks.